src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp

```diff
@@ -289,26 +439,54 @@ namespace OpenMS
       ///////////////////////////////////
 
       std::vector<double> normalized_library_intensity;
-      transition_group.getLibraryIntensity(normalized_library_intensity);
+      transition_group_detection.getLibraryIntensity(normalized_library_intensity);
       OpenSwath::Scoring::normalize_sum(&normalized_library_intensity[0], boost::numeric_cast<int>(normalized_library_intensity.size()));
-      std::vector<std::string> native_ids;
-      for (Size i = 0; i < transition_group.size(); i++)
+      std::vector<std::string> native_ids_detection;
+      for (Size i = 0; i < transition_group_detection.size(); i++)
       {
-        native_ids.push_back(transition_group.getTransitions()[i].getNativeID());
+        native_ids_detection.push_back(transition_group_detection.getTransitions()[i].getNativeID());
       }
 
       OpenSwath_Scores scores;
-      scorer.calculateChromatographicScores(imrmfeature, native_ids, normalized_library_intensity,
+      scorer.calculateChromatographicScores(imrmfeature, native_ids_detection, normalized_library_intensity,
                                             signal_noise_estimators, scores);
 
       double normalized_experimental_rt = trafo.apply(imrmfeature->getRT());
-      scorer.calculateLibraryScores(imrmfeature, transition_group.getTransitions(), *pep, normalized_experimental_rt, scores);
+      scorer.calculateLibraryScores(imrmfeature, transition_group_detection.getTransitions(), *pep, normalized_experimental_rt, scores);
       if (swath_map->getNrSpectra() > 0 && su_.use_dia_scores_)
       {
-        scorer.calculateDIAScores(imrmfeature, transition_group.getTransitions(),
+        scorer.calculateDIAScores(imrmfeature, transition_group_detection.getTransitions(),
                                   swath_map, ms1_map_, diascoring_, *pep, scores);
       }
 
+      if (su_.use_uis_scores && transition_group_identification.getTransitions().size() > 0)
```

Comment:
what happens if the user wants to use uis scores but some groups do not have identification transitions (e.g. `transition_group_identification.getTransitions().size() == 0)`
I think what happensis you call `getMetaValue` in OpenSwathWorkflow which will return an empty `DataValue`. When casting this to a String, the code will throw an exception `Exception::ConversionError`, see https://github.com/grosenberger/OpenMS/blob/feature/uisscoring/src/openms/include/OpenMS/DATASTRUCTURES/DataValue.h#L123

so there probably should be some code to handle this case, either here or in OpenSwathWorkflow